r/README.md

```diff
@@ -103,6 +103,7 @@ cmake
   -DARROW_DATASET=ON \
   -DARROW_FILESYSTEM=ON \
   -DARROW_JEMALLOC=ON \
+  -DARROW_MIMALLOC=ON \
```

Comment:
Maybe list this as optional below

Reply:
Done in a59b9a2. I noticed the S3 flag wasn't in the list of optional flags here so I added that as well